src/Server.zig

```diff
@@ -755,6 +758,52 @@ fn handleConfiguration(server: *Server, json: std.json.Value) error{OutOfMemory}
     };
 }
 
+fn handleShowBuildRunnerFailure(server: *Server, millis: i64, json: std.json.Value) Error!void {
+    const res = std.json.parseFromValue(?types.MessageActionItem, server.allocator, json, .{}) catch return error.ParseError;
+    defer res.deinit();
+
+    server.build_runner_failures_lock.lock();
+    defer server.build_runner_failures_lock.unlock();
+
+    const stderr = (server.build_runner_failures.fetchRemove(millis) orelse return).value;
+    defer server.allocator.free(stderr);
+
+    if (res.value != null) {
+        const request_id = try std.fmt.allocPrint(
+            server.allocator,
+            "show_document-{d}",
+            .{millis},
+        );
+
+        const cache = std.fs.cwd().makeOpenPath("zig-cache", .{}) catch return;
+        cache.writeFile2(.{
+            .sub_path = "build_runner_failure.txt",
```

Comment:
Thus hurts me lol; iirc `rust-analyzer` does this via per-editor integration which means they can use virtual documents rather than real ones. I put this in `zig-cache` rather than `zig-cache/tmp` because VSCode opens the target directory if it's in-workspace. The system tmp directory could be more fruitful, but I didn't have the time to implement that. Lemme know and I'll happily do it :)